testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/adminonly/WebConsoleRedirectionTestCase.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ *
+ */
+
+package org.jboss.as.test.manualmode.adminonly;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import io.undertow.util.Headers;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import javax.inject.Inject;
+
+import org.apache.http.client.methods.HttpGet;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ * @author Tomas Hofman (thofman@redhat.com)
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class WebConsoleRedirectionTestCase {
+
+    @Inject
+    protected static ServerController controller;
+
+    @Test
+    public void testRedirectionNoConsoleInAdminMode() throws Exception {
+        controller.startInAdminMode();
+
+        final HttpURLConnection connection = getConnection();
+        connection.setRequestMethod(HttpGet.METHOD_NAME);
+        connection.connect();
+        assertEquals(HttpURLConnection.HTTP_MOVED_TEMP, connection.getResponseCode());
```

Comment:
The fact this passes makes me suspect something is wrong. Without the change to ManagementRootConsoleRedirectHandler what response code is returned? A 404 or a 301?

If it's a 301 there's a separate issue that we'd need to sort out. There is no console possible in a simple WildFly Core server. The console module isn't even present. A redirect to a page that talks about admin-only is not valid.

If it's a 404, then we're good. The test of admin-only would be in WildFly full which is where we ship a console.

Reply:
It returns 301 before the change. I tried starting the container in normal mode instead of admin-only, and that returns 404 - no console. I will look into why it gives different results.

So the test case would have to go into wildfly testsuite. Probably best if I create a separate WFLY issue then.

Thanks for feedback!

Reply:
Thanks. My guess is the logic that decides what ConsoleHandler to use is checking things in the wrong order.

Reply:
Okey, so first the BaseHttpInterfaceAddStepHandler sets consoleEnabled attribute to true [1], because that's the default value for the attribute (console-enabled is undefined in the model).

After that, HttpManagementAddHandler selects either ConsoleMode.CONSOLE or ConsoleMode.ADMIN_ONLY [2], based on whether we are in admin-only or not.

If ADMIN_ONLY is selected, it will simply be used to create redirect handler with 302 (which however is not used in current code, hence this PR).

But, if ConsoleMode.CONSOLE was selected, handler creation will fail because of "ModuleNotFoundException: WFLYDMHTTP0007: No console module available with module name org.jboss.as.console.main" [3]. The console handler therefore stays null, which leads to 404 being returned [4] eventually.

Can we set console-enabled to false in default standalone.xml configuration file? I think it's generated in core-feature-pack module. Would that affect wildfly full somehow?

[1] https://github.com/wildfly/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/management/BaseHttpInterfaceAddStepHandler.java#L75
[2] https://github.com/wildfly/wildfly-core/blob/master/server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java#L204
[3] https://github.com/wildfly/wildfly-core/blob/master/domain-http/interface/src/main/java/org/jboss/as/domain/http/server/ManagementHttpServer.java#L350
[4] https://github.com/wildfly/wildfly-core/blob/master/domain-http/interface/src/main/java/org/jboss/as/domain/http/server/ManagementRootConsoleRedirectHandler.java#L61